room/compiler-processing/src/main/java/androidx/room/compiler/processing/ksp/KspTypeElement.kt

```diff
@@ -251,6 +252,40 @@ internal sealed class KspTypeElement(
         return declaration.classKind == ClassKind.OBJECT
     }
 
+    override fun isCompanionObject(): Boolean {
+        return declaration.isCompanionObject
+    }
+
+    override fun isAnnotationClass(): Boolean {
+        return declaration.classKind == ClassKind.ANNOTATION_CLASS
+    }
+
+    override fun isClass(): Boolean {
+        return declaration.classKind == ClassKind.CLASS
+    }
+
+    override fun isDataClass(): Boolean {
+        return Modifier.DATA in declaration.modifiers
+    }
+
+    override fun isValueClass(): Boolean {
+        return Modifier.INLINE in declaration.modifiers
+    }
+
+    override fun isFunctionalInterface(): Boolean {
+        // TODO: Update this once KSP supports it
+        // https://github.com/google/ksp/issues/393
+        env.messager.printMessage(
```

Comment:
is this the kind of logging you meant?

Reply:
lgtm. it will end up in logs in the user's compilation but then it would bubble back to the developer. for now, better be safe & verbose.